components/omega/src/infra/Field.cpp

```diff
@@ -0,0 +1,983 @@
+// Checks for the existence of a metadata entry with the given name
+bool Field::hasMetadata(
+    const std::string &MetaName // [in] Name of metadata entry
+) const {
+   return (FieldMeta.find(MetaName) != FieldMeta.end());
+}
+
+//------------------------------------------------------------------------------
+// Adds a metadata entry with the (name,value) pair
+int Field::addMetadata(const std::string &MetaName, // [in] Name of new metadata
+                       const std::any Value         // [in] Value of metadata
+) {
+   int RetVal = 0;
+
+   if (hasMetadata(MetaName)) {
+      LOG_ERROR("Failed to add the metadata {} to field {} because the field "
+                "already has an entry with that name.",
+                MetaName, FldName);
+
+      RetVal = -1;
+
+   } else {
+      FieldMeta[MetaName] = Value;
+   }
+
+   return RetVal;
+}
+
+//------------------------------------------------------------------------------
+// Adds multiple metadata entries with a list of (name,value) pairs
+int Field::addMetadata(
+    const std::initializer_list<std::pair<std::string, std::any>> &MetaPairs) {
+
+   int RetVal = 0;
+
+   // Loop through pairs and add each individually
+   for (const auto &MetaPair : MetaPairs) {
+      RetVal += addMetadata(MetaPair.first, MetaPair.second);
+   }
+
+   return RetVal;
+}
+
+//------------------------------------------------------------------------------
+// Removes a metadata entry with the given name
+int Field::removeMetadata(
+    const std::string &MetaName // [in] Name of metadata entry to remove
+) {
+
+   int RetVal = 0;
+
+   // Make sure metadata exists
+   if (hasMetadata(MetaName)) {
+      if (FieldMeta.erase(MetaName) != 1) {
+         LOG_ERROR("Unknown error removing metadata {} in field {} ", MetaName,
+                   FldName);
+         RetVal = -2;
+      }
+
+   } else {
+      LOG_ERROR("Failed to remove metadata {} for the field {}: "
+                "entry does not exist.",
+                MetaName, FldName);
+      RetVal = -1;
+   }
+
+   return RetVal;
+
+} // end removeMetadata
+
+//----------------------------------------------------------------------------//
+// Removes all defined Metadata
+void Field::removeAllMetadata() { FieldMeta.clear(); }
+
+//------------------------------------------------------------------------------
+// Retrieves the value of the metadata associated with a given name
+// This specific version of the overloaded interface coerces the value
+// to an I4 type
+int Field::getMetadata(
+    const std::string &MetaName, // [in] Name of metadata to get
+    I4 &Value                    // [out] I4 Value of metadata
+) {
+
+   int RetVal = 0;
+
+   if (hasMetadata(MetaName)) {
+      Value = std::any_cast<I4>(FieldMeta[MetaName]);
+   } else {
+      LOG_ERROR("Metadata {} does not exist for field {}.", MetaName, FldName);
+      RetVal = -1;
+   }
+
+   return RetVal;
+}
```

Comment:
Yes, since there weren't that many interfaces, I used aliasing rather than templating to provide a cleaner interface and avoid the extra <type> argument.